pkg/controller/operators/olm/requirements.go

```diff
@@ -95,6 +103,16 @@ func (a *Operator) requirementStatus(strategyDetailsDeployment *v1alpha1.Strateg
 			continue
 		}
 
+		if names := othersInstalledAlongside(crd, csv); len(names) > 0 && ownedCRDNames[crd.Name] {
+			if replacement := a.isBeingReplaced(csv, a.csvSet(csv.GetNamespace(), v1alpha1.CSVPhaseAny)); replacement == nil {
```

Comment:
That's intentional, not to prevent noise, but to avoid cases where the CSV being replaced transitions from Replacing to Pending (RequirementsNotMet) in the middle of an upgrade. This happens sometimes today and requires manual intervention (usually deleting the old CSV) to unblock the upgrade, even though the upgrade would otherwise have completed successfully.

Reply:
On re-read, you're right, good catch. There are at least a couple issues with replacement chain handling that need test cases and updates to this test. Thanks!

Reply:
To be more clear, consider the following CSVs in namespace `ns`:
- A, A', and B all claim the same CRD
- A' replaces A

In this case, it looks like A would **not** become pending even though B, with no relation, exists. Do we care about this?

Reply:
Discussed this at length last week. Ultimately, this test is never reached if there are multiple CRD owners in separate replacement chains within the same namespace, but I've updated it to explicitly consider only ancestors within the reconciling CSV's replacement chain.